Sources/Widgets/Widgets3D/LineWidget/index.js

```diff
@@ -132,39 +145,20 @@ function vtkLineWidget(publicAPI, model) {
   // --- Public methods -------------------------------------------------------
 
   publicAPI.getDistance = () => {
-    const nbHandles =
-      model.widgetState.getHandle1List().length +
-      model.widgetState.getHandle2List().length;
-    if (nbHandles !== 2) {
+    const nbHandles = model.widgetState.getNbHandles();
+    if (nbHandles < 2) {
       return 0;
     }
     return Math.sqrt(
       distance2BetweenPoints(
-        model.widgetState.getHandle1List()[0].getOrigin(),
-        model.widgetState.getHandle2List()[0].getOrigin()
+        model.widgetState.getHandle1().getOrigin(),
+        model.widgetState.getHandle2().getOrigin()
       )
     );
   };
 
-  publicAPI.updateTextValue = (text) => {
-    if (typeof model.widgetState.getText() !== 'undefined')
-      model.widgetState.getText().setText(text);
-  };
-
-  publicAPI.updateTextProps = (input, prop) => {
-    if (prop === 'positionOnLine') {
-      publicAPI.setPositionOnLine(input / 100);
-    }
-    updateTextPosition(model, publicAPI.getPositionOnLine());
-    model.widgetState.setPositionOnLine(publicAPI.getPositionOnLine());
-  };
-
-  publicAPI.updateHandleFromUI = (input, handleId) => {
-    if (handleId === 1) {
-      model.handle1Shape = input;
-    } else if (handleId === 2) {
-      model.handle2Shape = input;
-    }
+  publicAPI.setHandleShape = (handleId, shape) => {
+    model[`handle${handleId}Shape`] = shape;
     initializeHandleRepresentations();
     publicAPI.getRepresentationsForViewType(0);
```

Comment:
Why do you need it ? I do not see how a "getter" can change/set anything

Reply:
this function is used in example to change handles from UI.

Reply:
Truth is, if I comment it, it still works in the example. So my understanding is it should be removed. Do not keep useless function